### ISSUE TITLE
When rerunning a task, it doesn't take dependencies into account

### ISSUE THREAD

**Issue by Eijebong** [taskcluster/taskcluster#5442]
Scenario:

I have an `A` task that generates an artifact used by a `B` task. `B` depends on `A`.
I want to rerun `A` then `B` because some environment changed so the generated artifacts is different.
I click rerun on `A` then on `B`.
`B` runs without waiting for the new `A` run to finish.

**Comment by lotas** [comment]
Just to clarify, was the first time it run, `A` was successful? 

So in other words, dependency check should be against the most recent run. We probably don't reset task state once we rerun it.

**Comment by Eijebong** [comment]
Yeah, first `A` run was successful. The UI shows the right state though
